Feat/filter in request (#266)
socotecio/django-socio-grpc: docs/settings.rst

.. code-block:: python

  "REQUIRE_CLIENT_AUTH": True

.. _settings-filter-behavior:

FILTER_BEHAVIOR
^^^^^^^^^^^^^^^^^^^

Variable allowing user to configure how the filter work.

Values possible are: 

- ``METADATA_STRICT`` that only allow filtering against metadata. This is the default behavior as it's the legacy way of filtering. This will change for 1.0.0. This mode doesn't add additional field into the messages
- ``REQUEST_STRUCT_STRICT`` that only allow filtering against request ``_filters`` field. This mode add the ``_filters`` key in every message except if service specifically disable it.
- ``METADATA_AND_REQUEST_STRUCT`` that allow filtering against metadata and request ``_filters`` field. This mode add the ``_filters`` key in every message except if service specifically disable it. If filter is present in both metadata and _filters field, the one in _filters fiels have priority

> The mains differences between metadata and request are:
> - Metadata are not specified in the proto file so you add or removing filtering possibility without deploying a new api verison or causing breaking change
> - Request allow you to help the developper understand which endpoint accept filtering and automatically document it
> - Request are serialized so it can improve performance if filtering with large amount of data
> 
> An util class exist to help you set this settings:
> 
> .. code-block:: python

Reviewer: ```suggestion
The mains differences between metadata and request are:
- Metadata are not specified in the proto file so adding or removing a filter in a new version is not documented and may cause breaking changes.
- Requests allows helping developpers understand which endpoint accept filtering and automatically document it
- Requests are serialized so it can improve performance when filtering large amount of data

.. code-block:: python
```

.. code-block:: python

  "REQUIRE_CLIENT_AUTH": True

.. _settings-filter-behavior:

FILTER_BEHAVIOR
^^^^^^^^^^^^^^^

.. warning::
  
  Default behavior will change in 1.0.0 to accept only request filtering. If you start a new project please consider setting FILTER_BEHAVIOR to REQUEST_STRUCT_STRICT

Variable allowing user to configure how the filter work.

> The differents options possibles are described in :func:`django_socio_grpc.settings.FilterAndPaginationBehaviorOptions` 

Reviewer: ```suggestion
The differents options are described in :func:`django_socio_grpc.settings.FilterAndPaginationBehaviorOptions` 
```

.. code-block:: python

  from django_socio_grpc.settings import FilterAndPaginationBehaviorOptions

  "FILTER_BEHAVIOR": FilterAndPaginationBehaviorOptions.METADATA_STRICT

.. _settings-pagination-behavior:

PAGINATION_BEHAVIOR
^^^^^^^^^^^^^^^^^^^

.. warning::
  
  Default behavior will change in 1.0.0 to accept only request filtering. If you start a new project please consider setting PAGINATION_BEHAVIOR to REQUEST_STRUCT_STRICT

Variable allowing user to configure how the pagination work.

> The differents options possibles are described in :func:`django_socio_grpc.settings.FilterAndPaginationBehaviorOptions` 

Reviewer: ```suggestion
The differents options are described in :func:`django_socio_grpc.settings.FilterAndPaginationBehaviorOptions` 
```

.. code-block:: python

  from django_socio_grpc.settings import FilterAndPaginationBehaviorOptions
  
  "PAGINATION_BEHAVIOR": FilterAndPaginationBehaviorOptions.METADATA_STRICT

.. _settings-default-message-name-constructor:

DEFAULT_MESSAGE_NAME_CONSTRUCTOR
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Variable that indicate the class used to generate the name of the proto messages. Default is :func:`MessageNameConstructor <django_socio_grpc.protobuf.message_name_constructor.MessageNameConstructor>`.

> For more informations see :ref:`the specific associated documentation <proto-generation-message-name-constructor>`

Reviewer: ```suggestion
For more informations see :ref:`the documentation <proto-generation-message-name-constructor>`
```

.. code-block:: python
  
  "DEFAULT_MESSAGE_NAME_CONSTRUCTOR": "django_socio_grpc.protobuf.message_name_constructor.MessageNameConstructor"

.. _settings-default-generation-plugin:

DEFAULT_GENERATION_PLUGINS
^^^^^^^^^^^^^^^^^^^^^^^^^^

List of class inherited from :func:`BaseGenerationPlugin <django_socio_grpc.protobuf.generation_plugin.BaseGenerationPlugin>`
to specify plugin that are used globally for all actions.

> :ref:`See specific documentation <proto-generation-plugins>`

Reviewer: ```suggestion
:ref:`See documentation <proto-generation-plugins>`
```
